<?xml version="1.0"?>

<launch>
    <arg name="rviz_config" default="traj_opt_sub"/>
    <arg name="rviz" default="false"/>

    <param name="use_sim_time" value="false"/>

    <node name="traj_opt"
          pkg="trajectory_optimization"
          type="trajectory_optimization_sub.py" output="screen">

          <param name="point_cloud_topic" type="string" value="/final_cost_cloud" />
          <param name="path_topic" type="string" value="/path" />
          <param name="opt_steps" type="int" value="50" />
          <param name="smooth_weight" type="double" value="14.0" />
          <param name="length_weight" type="double" value="0.02" />
          <param name="lr_pose" type="double" value="0.2" />
          <param name="lr_quat" type="double" value="0.0" />

    </node>

    <node if="$(arg rviz)"
          name="rviz" pkg="rviz" type="rviz"
          args="-d $(find trajectory_optimization)/config/$(arg rviz_config).rviz"/>
</launch>
